### DISCUSSION TITLE
Get precise detection of objects in global NED frame

### DISCUSSION THREAD

**Discussion by qcsui** [microsoft/AirSim#4298] [Support Q&A]
Hi, right now I am trying to get coordinates of objects in global NED frame. For that, I used 
```py
cylinders = client.simGetDetections(camera_name, image_type)  # relative position in local NED frame
multirotor_state = client.getMultirotorState()  # drone's position in global NED frame
```
then by adding these two, I should get the object's location in the global NED frame. However, due to a time lag between these two actions(I think so), the position of targets that I got is not accurate.
![2022-01-18 112544](https://user-images.githubusercontent.com/40121130/149919477-98f5d83e-eb58-406f-a214-7b4f124d9582.jpg)
The big circle is my trajectory, and the triangles are the observations of targets.
So the questions are:
- Can anyone give me some suggestions on how I can get the exact position of the targets in the global NED frame?
- I noticed that `GeoPoint` is given in `DectionInfo` which should be able to convert into NED frame. Anybody has done that in practice?

**Comment by zimmy87** [comment]
Looking at the source code [here](https://github.com/microsoft/AirSim/blob/master/Unreal/Plugins/AirSim/Source/WorldSimApi.cpp#L1050), `DetectionInfo.GeoPoint` is derived directly from the object's ActorLocation, so this will yield the exact position of the target. In C++, you can use the msr::airlib::EarthUtils::GeodeticToNed API to convert this to NED frame, but this API isn't currently exposed in the python client. You could add this to the python client, but it may be easier to just use the `simGetObjectPose` API in python. The object name passed to `simGetObjectPose` should be the same as `DetectionInfo.name`.
